sdk/core/azure-core/src/main/java/com/azure/core/implementation/util/HttpUtils.java

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.core.implementation.util;
+
+/**
+ * Utilities shared with HttpClient implementations.
+ */
+public final class HttpUtils {
+    /**
+     * Context key used to indicate to an HttpClient implementation if it should eagerly read the response from the
+     * network.
+     */
+    public static final String AZURE_EAGERLY_READ_RESPONSE = "azure-eagerly-read-response";
+
+    /**
+     * Context key used to indicate to an HttpClient implementation if the response body should be ignored and eagerly
+     * drained from the network.
+     */
+    public static final String AZURE_IGNORE_RESPONSE_BODY = "azure-ignore-response-body";
+
+    /**
+     * Context key used to indicate to an HttpClient a per-call response timeout.
+     */
+    public static final String AZURE_RESPONSE_TIMEOUT = "azure-response-timeout";
```

Comment:
let's use the same terminology as for env vars and call it `azure-request-response-timeout`?

https://github.com/Azure/azure-sdk-for-java/tree/main/sdk/core/azure-core#http-timeouts

Also, I think we should mention new way to set timeout in the readme